apollo/smoothie.py

```diff
@@ -74,6 +74,10 @@ def send(self, gcode, response_handler=None):
             back as the second argument if you want it to handle responses 
             from the SmoothieBoard.
         """
+        if not isinstance(gcode, str):
+            logger.error('Gcode must be string')
+            raise TypeError('Gcode must be string')
+            return None
```

Comment:
Thanks, I was wondering about that
On Jun 14, 2016 12:28 PM, "SimplyAhmazing" <notifications@github.com> wrote:

> In apollo/smoothie.py
> <https://github.com/OpenTrons/sandbox-driver/pull/4#discussion_r67005688>:
>
> > @@ -74,6 +74,10 @@ def send(self, gcode, response_handler=None):
> >              back as the second argument if you want it to handle responses
> >              from the SmoothieBoard.
> >          """
> > +        if not isinstance(gcode, str):
> > +            logger.error('Gcode must be string')
> > +            raise TypeError('Gcode must be string')
> > +            return None
>
> raise ends program execution, the return None won't be reached...
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/OpenTrons/sandbox-driver/pull/4/files/8fdfbb476120cead0d128b223619bfaa99c99235..46bc44f425710aa6b5dca6e0ddc6ddeec83eaddc#r67005688>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AEzzl2JjkVc3pXFfdCQSkt_wnIrigebEks5qLtaRgaJpZM4IzA3_>
> .
>